src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -123,20 +132,51 @@ static String generateAppInstallationToken(String appId, String appPrivateKey, S
                 appInstallation = appInstallations.get(0);
             } else {
                 appInstallation = appInstallations.stream()
-                        .filter(installation -> installation.getAccount().getLogin().equals(owner))
-                        .findAny()
-                        .orElseThrow(() -> new IllegalArgumentException(String.format(ERROR_NOT_INSTALLED, appId)));
+                    .filter(installation -> installation.getAccount().getLogin().equals(owner))
+                    .findAny()
+                    .orElseThrow(() -> new IllegalArgumentException(String.format(
+                        ERROR_NOT_INSTALLED,
+                        appId)));
             }
 
             GHAppInstallationToken appInstallationToken = appInstallation
-                    .createToken(appInstallation.getPermissions())
-                    .create();
-
-            return appInstallationToken.getToken();
-        } catch (IOException e) {
+                .createToken(appInstallation.getPermissions())
+                .create();
+
+            long expiration = getExpirationSeconds(appInstallationToken);
+            LOGGER.log(Level.FINEST,
+                "Token raw expiration epoch seconds: {0,number,#}",
+                expiration);
+
+            AppInstallationToken token = new AppInstallationToken(appInstallationToken.getToken(),
+                expiration);
+            LOGGER.log(Level.FINE,
+                "Generated App Installation Token for app ID {0}",
+                appId);
+
+            return token;
+        } catch (Exception e) {
+            LOGGER.log(Level.WARNING, "Failed to retrieve GitHub App installation token for app ID " + appId, e);
```

Comment:
nit: I would either log the exception or throw it, but not both. Whatever ends up catching the thrown exception should log it and/or handle it, and you don't want duplicate log messages for the same exception.

Reply:
Unfortunately, it looks like some callers simply swallow the exception, maybe only logging it to their own logger but not to anywhere visible.  Also, when this fails this is useful tracing.  Maybe make it a `FINE`, but that also seems misleading.

Reply:
> some callers simply swallow the exception

So…fix them?

Reply:
@jglick  Out of scope for this PR and lower priority that any number of other tasks.

Reply:
If you _know_ that some callers swallow it then you must know which, right? So it should be a one-line patch to fix that? Confused.